apis/kueue/v1beta1/clusterqueue_types.go

```diff
@@ -0,0 +1,308 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterQueueSpec defines the desired state of ClusterQueue
+type ClusterQueueSpec struct {
+	// resourceGroups describes groups of resources.
+	// Each resource group defines the list of resources and a list of flavors
+	// that provide quotas for these resources.
+	// Each resource and each flavor can only form part of one resource group.
+	// resourceGroups can be up to 16.
+	// +listType=atomic
+	// +kubebuilder:validation:MaxItems=16
+	ResourceGroups []ResourceGroup `json:"resourceGroups,omitempty"`
+
+	// cohort that this ClusterQueue belongs to. CQs that belong to the
+	// same cohort can borrow unused resources from each other.
+	//
+	// A CQ can be a member of a single borrowing cohort. A workload submitted
+	// to a queue referencing this CQ can borrow quotas from any CQ in the
+	// cohort. Only quota from [resource, flavor] pairs listed in the CQ can be
+	// borrowed.
```

Comment:
```suggestion
	// cohort. Only borrowingLimit from [resources, flavors] pairs listed in the CQ can be
	// borrowed.
```

Reply:
The word quota was intended. The documentation for how much quota is allowed to be borrowed is left for the description of the `borrowingLimit` field itself.

Reply:
It sounds good to me.